bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/ICalendarHandlerFactory.java

```diff
@@ -56,6 +56,11 @@ public ICalendarHandlerFactory(@Reference HttpClientFactory httpClientFactory,
             @Reference EventPublisher eventPublisher) {
         this.eventPublisher = eventPublisher;
         sharedHttpClient = httpClientFactory.createHttpClient(BINDING_ID);
+        try {
+            sharedHttpClient.start();
+        } catch (Exception e) {
+            logger.debug("Exception while starting HttpClient.", e);
+        }
```

Comment:
@openhab/add-ons-maintainers since there is no way to recover from this error and we need a started client, wouldn't it be better to add `throws Exception` to throw an exception and let the framework retry starting the bundle in case of a failure?

Reply:
Was there a reason `getCommonHttpClient()` was not used?

Reply:
Good point. I think now. I thought that we require re-configuration of the client, but it seems that is not the case.

Reply:
Ah, so i understood the [sentence](https://www.openhab.org/docs/developer/guidelines.html#default-libraries) about the shared HttpClient wrong. I thought we should share the Client between Handlers in the Binding. I'll move to `getCommonHttpClient()` which also will be started.